### PR TITLE
Drop country middleware

### PR DESCRIPTION
**Pull request by maarcingebala**
This PR drops the country middleware (`saleor.core.middleware.country`), which was using GeoIP to determine the user's country. In the new approach, user location is provided as an argument in queries or mutations that need it for tax or stock calculations.

Note: this PR removes `CheckoutPaymentCreate.billingAddress` input field which is a breaking change. The idea is that we should always use billing address from the checkout instance, to have one way of providing billing addresses for payments. This field will be marked as deprecated in Saleor 2.11.

Fixes https://github.com/mirumee/saleor/issues/4315

Todo:
- [x] update test_variant_availability.py tests to use new address parameter
- [x] add tests for queries that use the new param
- [x] deprecate `CheckoutPaymentCreate.billingAddress` input on 2.11

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [x] Deprecated API fields or mutations
* [x] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1064 Add "country" input field in storefront's product queries (drop GeoIP)](https://app.clickup.com/t/2549495/SALEOR-1064)

File: saleor/graphql/utils/__init__.py [hunk 1]
```diff
@@ -131,3 +138,28 @@ def get_user_or_app_from_context(context):
...
+    """Get country of the current user to use for tax and stock related calculations.
+
+    User's country context is determined from the provided `destination_address` (which
+    may represent user's current location determined by the client or a shipping
+    address provided in the checkout). If `destination_address` is not given, the
+    default company address from the shop settings is assumed. If this address is not
+    set, fallback to the `DEFAULT_COUNTRY` setting.
+    """
+    if destination_address and destination_address.country:
+        if isinstance(destination_address, Country):
```
Comment (fowczarek) [inline saleor/graphql/utils/__init__.py:158]: It shouldn't be like this?
```suggestion
        if isinstance(destination_address.country, Country):
```

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/utils/__init__.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/utils/__init__.py:158]: Yep, it's a bug.

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/utils/__init__.py [hunk 2]
```diff
@@ -131,3 +138,28 @@ def get_user_or_app_from_context(context):
...
+            return destination_address.country.code
+        else:
+            return destination_address.country
+    elif company_address and company_address.country:
+        return company_address.country
+    else:
```
Comment (korycins) [inline saleor/graphql/utils/__init__.py:163]: `else` not needed :P

File: saleor/graphql/utils/__init__.py [hunk 3]
```diff
@@ -131,3 +138,28 @@ def get_user_or_app_from_context(context):
...
+    User's country context is determined from the provided `destination_address` (which
+    may represent user's current location determined by the client or a shipping
+    address provided in the checkout). If `destination_address` is not given, the
+    default company address from the shop settings is assumed. If this address is not
+    set, fallback to the `DEFAULT_COUNTRY` setting.
+    """
+    if destination_address and destination_address.country:
+        if isinstance(destination_address, Country):
+            return destination_address.country.code
+        else:
```
Comment (korycins) [inline saleor/graphql/utils/__init__.py:159]: `else` not needed

**Review by fowczarek — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Comment by maarcingebala** [comment]
PR is currently blocked by storefront/SDK which currently uses the `billingAddress` field that this PR removes from API.

**Review by patrys — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**